src/solvers/dg_unstructured_quad/dg_2d.jl

```diff
@@ -37,26 +38,26 @@ function rhs!(du::AbstractArray{<:Any,4}, u, t,
   @timeit_debug timer() "reset ∂u/∂t" du .= zero(eltype(du))
 
   # Calculate volume integral
-  @timeit_debug timer() "volume integral" calc_contravariant_volume_integral!(du, u, have_nonconservative_terms(equations), equations,
-                                                                              dg.volume_integral, dg, cache)
+  @timeit_debug timer() "volume integral" calc_volume_integral!(du, u, have_nonconservative_terms(equations), equations,
+                                                                dg.volume_integral, dg, cache, mesh)
```

Comment:
```suggestion
  @timeit_debug timer() "volume integral" calc_volume_integral!(du, u, mesh, have_nonconservative_terms(equations), equations,
                                                                dg.volume_integral, dg, cache)
```
We usually have the order `mesh, equations, solver, cache`. That's also the signature the BFG group uses in their structured stuff. I strongly prefer keeping the same order here (and at other similar places).

Reply:
okay, just a consistency note ;) `prolong2boundaries!` in the tree version `solvers/dg/dg_2d.jl` puts the cache first and the BFG guys do not have a `prolong2boundaries!` routine. So I will use the ordering `mesh, equations, solver, cache` for internal consistency within unstructured quad

Reply:
Yeah, that's conflicting with the general rule to put the (main) modified arguments first.

Reply:
... and causing unending headaches for me, since my mind is in eternal dissonance over these two non-orthogonal rules.

Reply:
That's true... Shall we just prioritize consistency and enforce the order `mesh, equations, solver, cache` everywhere?

Reply:
I don't know... what if we have `foo!(cache, solver)` where clearly `cache` is modified? Should this then be also reordered? Probably not. But `foo!(cache, equations, solver)` one could already argue it makes sense. Thus would we enforce this rule whenever we see one of the four appearing? What about `bar!(baz, mesh, equations, cache)` where `cache` and `baz` are modified - wouldn't that be super confusing that `baz` is in the beginning, but `cache` is not? Or should `baz` then go to the end?

For the time being at least, I say we leave the rule hierarchy as
1. modified argument first
2. whenever `mesh`, `equations`, `solver`, and `cache` (or a subset thereof) appear in a function argument list, put them in this order, unless it violates 1.

Reply:
Well, the `cache` is (or at least can be) modified in many functions but is still passed as last argument (since it just provides some additional context but it not that important). For example, `calc_volume_integral!(du, u, ..., cache)` can modify the `cache`, e.g. the fields `element_ids_dg, element_ids_dgfv` for shock-capturing stuff. But you're right, we should agree on a specific rule. My basic choice was
1. The main modified argument comes first. For example, if we modify `du`, it should come first. If we only modify the `cache` as in `prolong2interfaces!` and its siblings, put the `cache` first.
2. Otherwise, use the order `mesh, equations, solver, cache`.
3. If something needs to be specified more for dispatch, put the additional argument before the one that is specified in more detail. For example, we use `have_nonconservative_terms(equations), equations` and `dg.mortar, dg`

Reply:
Maybe we should add this to `conventions.md`?

Reply:
okay, I can modify my input orderings according to this

Reply:
> Maybe we should add this to `conventions.md`?

Done in https://github.com/trixi-framework/Trixi.jl/pull/544 (but to a different place where it fits better IMHO)